File: Javascript/ReverseLL.js
// Reverses a singly linked list

class ListNode {
    constructor(val = 0, next = null) {
        this.val = val;
        this.next = next;
    }
}

class ReverseLL {
    reverseList(head) {
        let prev = null;
        let cur = head;

        while (cur) {
            let nextNode = cur.next;
            cur.next = prev;
            prev = cur;
            cur = nextNode;
        }
        return prev;
    }

    printList(head) {
        let cur = head;
        let output = "";
        while (cur) {
            output += cur.val + " ";
            cur = cur.next;
        }
        console.log(output.trim());
    }
}

(function main() {
    let head = new ListNode(1, new ListNode(2, new ListNode(3)));

    const llUtils = new ReverseLL(); 

    console.log("Original:");
    llUtils.printList(head); 

    head = llUtils.reverseList(head); 

    console.log("Reversed:");
    llUtils.printList(head); 
})();

File: Java/ReverseLL.java
// Reverses a singly linked list

class ListNode{
    int val;
    ListNode next;
    ListNode(int x) {val = x;}
}

public class ReverseLL {
    static ListNode reverseList(ListNode head) {
        ListNode prev = null;
        ListNode cur = head;

        while(cur != null){
            ListNode next = cur.next;
            cur.next = prev;
            prev = cur;
            cur = next;
        }
        return prev;
    }

    static void printList(ListNode head){
        while(head != null){
            System.out.print(head.val + " ");
            head = head.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        ListNode head = new ListNode(1);
        head.next = new ListNode(2);
        head.next.next = new ListNode(3);

        System.out.println("Original: ");
        printList(head);

        head = reverseList(head);

        System.out.print("Reversed: ");
        printList(head);
    }
}